### PR TITLE
Extract lib wx init

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #3482 
Resolves: #3492 

Extract libraries that are not toolkit neutral but encapsulate what is behind the BasicUI facade and will need corresponding reimplementation against Qt

QA:
- [ ] Test the file picker message box, including on Linux; test the error dialogs when opening non-existent file, confirmation when overwriting a file on save
- [x] Test screen readers on Windows 
- [x] Test text entry dialog, such as when making a new label with the Edit > Labels > Edit Labels... dialog
- [x] Test controls of different kinds in any effect dialog
- [x] Test some progress dialogs
- [x] Test the plugin registaration dialog
- [x] Test the Log window (Help menu)
- [x] Test the Error dialog -- as with the alpha-only crash-me buttons in Help menu

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
The dependency #4313 required rebasing onto recent master and conflict resolution in `#include` directives, and then this one did too, onto that rebasing.

**Comment by Paul-Licameli** [comment]
And rebased again.

File: libraries/lib-foo/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+did not yet provide std::make_unique.  That explains the name.
+
+Audacity now uses C++17, and yet, need arose for other anticipations of future
+standards.  It also provides other pervasively used utilities that don't
+correspond to things in the standard.
+
+]]#
+
+set( SOURCES
+   Foo.cpp
```
Comment (vsverchinsky) [inline libraries/lib-foo/CMakeLists.txt]: Not sure that I understand what this is for

File: libraries/lib-foo/CMakeLists.txt [hunk 2]
```diff
@@ -1,26 +0,0 @@
...
-did not yet provide std::make_unique.  That explains the name.
-
-Audacity now uses C++17, and yet, need arose for other anticipations of future
-standards.  It also provides other pervasively used utilities that don't
-correspond to things in the standard.
-
-]]#
-
-set( SOURCES
-   Foo.cpp
```
Comment (vsverchinsky) [inline libraries/lib-foo/CMakeLists.txt]: Ah, so this was indeed committed by mistake. I guess that this change should be squashed

File: libraries/lib-foo/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Historically, MemoryX.h began as a wrapper of C++11 <memory>, which by itself
+did not yet provide std::make_unique.  That explains the name.
+
+Audacity now uses C++17, and yet, need arose for other anticipations of future
+standards.  It also provides other pervasively used utilities that don't
+correspond to things in the standard.
+
+]]#
+
+set( SOURCES
```
Comment (crsib) [inline libraries/lib-foo/CMakeLists.txt]: lib-foo is still with us? I think we need lib-bar to, so it doesn't feel lonely xD

File: libraries/lib-foo/CMakeLists.txt [hunk 4]
```diff
@@ -1,26 +0,0 @@
-#[[
```
Comment (crsib) [inline libraries/lib-foo/CMakeLists.txt]: And now it's gone :-)

File: src/prefs/LibraryPrefs.cpp [hunk 5]
```diff
@@ -24,6 +24,17 @@ MP3 and FFmpeg encoding libraries.
 #include <wx/defs.h>
 #include <wx/stattext.h>
 
+<<<<<<< HEAD
```
Comment (vsverchinsky) [inline src/prefs/LibraryPrefs.cpp]: This diff shouldn't be there

Comment (crsib) [inline src/prefs/LibraryPrefs.cpp]: Is it expected?

File: src/prefs/LibraryPrefs.cpp [hunk 6]
```diff
@@ -24,17 +24,6 @@ MP3 and FFmpeg encoding libraries.
 #include <wx/defs.h>
 #include <wx/stattext.h>
 
-<<<<<<< HEAD
```
Comment (vsverchinsky) [inline src/prefs/LibraryPrefs.cpp]: Here it is removed, but would be better to squash as we always do

**Review by vsverchinsky — CHANGES_REQUESTED** [review]

**Comment by Paul-Licameli** [comment]
Sorry guys, I made mistakes rebasing this one, and did not clean up the intermediate commits correctly.  Expect another force push and I will invite review again.

**Review by vsverchinsky — APPROVED** [review]

**Comment by Paul-Licameli** [comment]
Rebased onto recent master and resolved conflicts

**Comment by Paul-Licameli** [comment]
A difficult conflict resolution with e892cea70e50e24dd168a6c13966b4a007761bca

Review again, see new inserted commit

**Comment by crsib** [comment]
The branch requires conflict resolution now

**Comment by Paul-Licameli** [comment]
I have rebased and resolved conflicts with the vertial ruler project.  Just a lot of `#include` lines and the list of files in src/CMakeLists.txt.

**Review by crsib — APPROVED** [review]

**Pull request merged**